app/src/main/java/co/copperhead/pdfviewer/PdfViewer.java

```diff
@@ -81,12 +85,8 @@ public void setNumPages(int numPages) {
 
         @JavascriptInterface
         public void setDocumentProperties(final String properties) {
-            if (mDocumentProperties != null) {
```

Comment:
I'd rather keep this sanity check, since we're meant to treat the JavaScript code as compromised here and there's no reason it should be calling this more than once that I can see. It makes a fresh instance for each document.

Reply:
Theoretically you are correct, but this throws an exception on each orientation change, due to the properties never being null (they are saved and restored in onCreate()).
Ideally this could be avoided, but due to a loader issue which is being triggered because the loader can't be initialized in onCreate(), the properties state must be saved, as the loader doesnt return the retrieved data after orientation change as it normally would.

There is an ugly workaround to this tough, if you want to keep the sanity check, which can be applied.